homeassistant/components/onvif/parsers.py

```diff
@@ -0,0 +1,358 @@
+"""ONVIF event parsers."""
+from homeassistant.util import dt as dt_util
+from homeassistant.util.decorator import Registry
+
+from .models import Event
+
+PARSERS = Registry()
+
+
+@PARSERS.register("tns1:VideoSource/MotionAlarm")
+# pylint: disable=protected-access
+async def async_parse_motion_alarm(uid: str, msg) -> Event:
+    """Handle parsing event message.
+
+    Topic: tns1:VideoSource/MotionAlarm
+    """
+    try:
+        source = msg.Message._value_1.Source.SimpleItem[0].Value
+        return Event(
+            f"{uid}_{msg.Topic._value_1}_{source}",
+            f"{source} Motion Alarm",
+            "binary_sensor",
+            "motion",
```

Comment:
The device class must be one of the constants from the binary sensor or sensor entity integrations.

Reply:
That’s correct. Was that just a note or is there a constant I should use here instead?

Reply:
We should import and use the constants. Here they are:
https://github.com/home-assistant/core/blob/36ad9a25e39b46e49a644903ce69cb0745f0faac/homeassistant/components/binary_sensor/__init__.py#L25-L95

Reply:
https://github.com/home-assistant/core/blob/36ad9a25e39b46e49a644903ce69cb0745f0faac/homeassistant/components/sensor/__init__.py#L9-L16